hivemind/utils/grpc.py

```diff
@@ -6,19 +6,44 @@
 import torch
 
 from hivemind.proto import runtime_pb2
+from hivemind.proto.runtime_pb2 import CompressionType
 
 
-def serialize_torch_tensor(tensor: torch.Tensor) -> runtime_pb2.Tensor:
+def serialize_torch_tensor(tensor: torch.Tensor, compression_type=CompressionType.NONE) -> runtime_pb2.Tensor:
     array = tensor.numpy()
-    proto = runtime_pb2.Tensor(
-        buffer=array.tobytes(),
-        size=array.shape,
-        dtype=array.dtype.name,
-        requires_grad=tensor.requires_grad)
+    if compression_type == CompressionType.MEANSTD_LAST_AXIS_FLOAT16:
+        assert array.dtype == np.float32
+
+        mean = array.mean()
+        std = array.std()
+        normalized = (array - mean) / std
```

Comment:
Request: you're using numpy mean/std, but torch also has its own mean/std functions. Can you please test which one is faster (on cpu)?

Reply:
Third option: make it a three-liner:
- compute mean
- subtract mean in-place
- compute std = euclidean norm, since we already subtracted mean
- divide by std